internal/run/convert.go

```diff
@@ -0,0 +1,190 @@
+package run
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/helm/helm-2to3/pkg/v3"
+	"github.com/helm/helm-2to3/pkg/common"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	utils "github.com/maorfr/helm-plugin-utils/pkg"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/pelotech/drone-helm3/internal/env"
+)
+
+type Convert struct {
+	*config
+	release string
+
+	deleteV2Releases bool
+	dryRun bool
+	kubeContext string
+	kubeConfig string
+	label string
+	releaseVersionsMax int
+	tillerNS string
+
+	cmd cmd
+}
+
+func NewConvert(cfg env.Config, tillerNS string, kubeConfig string) *Convert {
+	return &Convert{
+		config: newConfig(cfg),
+		release: cfg.Release,
+		deleteV2Releases: cfg.DeleteV2Releases,
+		dryRun: cfg.DryRun,
+		kubeContext: cfg.Context,
+		kubeConfig: kubeConfig,
+		label: cfg.TillerLabel,
+		releaseVersionsMax: cfg.ReleaseVersionsMax,
+		tillerNS: tillerNS,
+	}
+}
+
+// Execute runs the `2to3 convert` command from Prepare
+// it detects if a conversion is needed and if not it will return no error
+// if a conversion is required then any returned error is related to a step
+// in the conversion process
+func (c *Convert) Execute() error {
+
+	clientSet, err := clientSetFromKubeConfig(c.kubeConfig)
+	if err != nil {
+		return err
+	}
+
+	// If Tiller is not present, utils.ListReleasesWithKubeConfig will panic
+	if !isTillerPresent(clientSet, "name=tiller,app=helm") {
+		log.Println("There is no Tiller Deployment running")
+		return nil
+	}
+
+	// Check for existence of v2 Release (only if Tiller is running)
+	var v2ReleaseFound bool
```

Comment:
nit: this variable isn't necessary since you return early if no v2 releases are found